### PR TITLE
Add levels model

### PR DESCRIPTION
**Pull request by kxmbrian**

### CODE REVIEW

**Comment by kxmbrian** [comment]
@wangqiang1208 told me this has to be named-spaced under courses. Not ready to be merged yet

**Comment by lowjoel** [comment]
Please squash all noisy commits, e.g. 'fixing lints', then this is ready to be merged.

Do you have the design for this component somewhere? We should document it in the wiki.

**Comment by kxmbrian** [comment]
Is there is a standard on the wiki documentation that we should adopt?

**Comment by lowjoel** [comment]
Probably we can start with a skeleton, we can move it around later when we have more structures.

File: app/models/course/level.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class Course::Level < ActiveRecord::Base
+  belongs_to :course, inverse_of: :levels
+  validates :exp_threshold, numericality: { greater_than: 0 }
```
Comment (lowjoel) [inline app/models/course/level.rb]: based on the new design, please pick proper nouns to describe what's going on.

Comment (kxmbrian) [inline app/models/course/level.rb]: @lowjoel do you mean that for exp_threshold? I think it is the appropriate term already?

**Comment by kxmbrian** [comment]
This PR is messy. I'm closing this and creating a fresh one.

**Pull request closed without merging**